### ISSUE TITLE
'pod install' regenerates changes in Pods/Pods.xcodeproj/project.pbxproj every time

### ISSUE THREAD

**Issue by littlebobert** [CocoaPods/CocoaPods#4051]
If I run pod install it generates changes to the entire pbxproj every time.

In the following terminal session, I run pod install, commit the changes in the .pbxproj, run pod install and then have new changes in the .pbxproj.

I thought this was a dupe of https://github.com/CocoaPods/CocoaPods/issues/2116 but that’s supposedly a dupe of https://github.com/CocoaPods/CocoaPods/issues/1118 which is supposed to be fixed in 0.38.0beta2, but I’m running 0.38.2.

```
[~/Downloads/Storyhouse]$ git status
On branch experimental/lobbyboy
Your branch is up-to-date with 'origin/experimental/lobbyboy'.
nothing to commit, working directory clean
[~/Downloads/Storyhouse]$ pod install
Updating local specs repositories
Analyzing dependencies
Downloading dependencies
Using AFNetworking (2.5.1)
Using FLAnimatedImage (1.0.8)
Using Heap (2.1.3)
Using Instabug (4.1.1)
Using KILabel (1.0.1)
Using MBProgressHUD (0.9.1)
Using Mixpanel (2.8.1)
Using PINCache (2.0)
Using PINRemoteImage (1.1.2)
Using SVProgressHUD (1.1.2)
Using Stripe (5.1.2)
Generating Pods project
Integrating client project
Sending stats
[~/Downloads/Storyhouse]$ git status
On branch experimental/lobbyboy
Your branch is up-to-date with 'origin/experimental/lobbyboy'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   Pods/Pods.xcodeproj/project.pbxproj

no changes added to commit (use "git add" and/or "git commit -a")
[~/Downloads/Storyhouse]$ git commit -a -m "Updated pods"
[experimental/lobbyboy 612a939] Updated pods
 1 file changed, 1281 insertions(+), 1281 deletions(-)
[~/Downloads/Storyhouse]$ pod install
Updating local specs repositories
Analyzing dependencies
Downloading dependencies
Using AFNetworking (2.5.1)
Using FLAnimatedImage (1.0.8)
Using Heap (2.1.3)
Using Instabug (4.1.1)
Using KILabel (1.0.1)
Using MBProgressHUD (0.9.1)
Using Mixpanel (2.8.1)
Using PINCache (2.0)
Using PINRemoteImage (1.1.2)
Using SVProgressHUD (1.1.2)
Using Stripe (5.1.2)
Generating Pods project
Integrating client project
Sending stats
[~/Downloads/Storyhouse]$ git status
On branch experimental/lobbyboy
Your branch is ahead of 'origin/experimental/lobbyboy' by 1 commit.
  (use "git push" to publish your local commits)
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   Pods/Pods.xcodeproj/project.pbxproj

no changes added to commit (use "git add" and/or "git commit -a")
[~/Downloads/Storyhouse]$
```

**Comment by segiddins** [comment]
Can you please share the project? Otherwise this will be impossible to debug.

**Comment by littlebobert** [comment]
It reproduces with an empty project and our Podfile: https://www.dropbox.com/s/egy6jkyrylsdb35/CocoapodsTest.zip?dl=0

**Comment by yonaskolb** [comment]
I'm also experiencing this

**Comment by segiddins** [comment]
Thanks, I'll try and get around to investigating this this week.

**Comment by segiddins** [comment]
Just confirmed this is reproducible with that project.

**Comment by segiddins** [comment]
I have no idea why this is happening...

**Comment by nvasavikumar** [comment]
I'm also experiencing the same issue with version 0.38.2

**Comment by Kurry** [comment]
+1

**Comment by Kurry** [comment]
Errrrrr so annoying.

**Comment by maxmeyers** [comment]
This is fixed for me in 0.39.0.beta.5

**Comment by segiddins** [comment]
@maxmeyers seriously?!?!? Wow, well I can promise it wasn't on purpose!

**Comment by maxmeyers** [comment]
@segiddins Sorry, I may have spoken too soon, I tried it with the sample project above and it's still happening.

**Comment by rnpatel** [comment]
@segiddins any updates on this issue?  We display the git commit inside our app and it's now showing as "-dirty" when built from our CI server.

**Comment by segiddins** [comment]
@rnpatel as evidenced by the lack of recent comments on this issue, there are no updates, but we'd welcome a PR fixing it :)

**Comment by DanielTomlinson** [comment]
This appears to be fixed in 39.0

```
$ pod install --no-repo-update
Analyzing dependencies
Downloading dependencies
Using AFNetworking (2.6.0)
Using FLAnimatedImage (1.0.8)
Using Heap (2.1.5)
Using Instabug (4.1.4)
Using KILabel (1.0.1)
Using MBProgressHUD (0.9.1)
Using Mixpanel (2.8.2)
Using PINCache (2.0)
Using PINRemoteImage (1.1.2)
Using SVProgressHUD (1.1.3)
Using Stripe (5.1.2)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 9 dependencies from the Podfile and 11
total pods installed.
$ gs
## master
 M CocoapodsTest.xcodeproj/project.pbxproj
 M Podfile.lock
 M Pods/Manifest.lock
 M Pods/Pods.xcodeproj/project.pbxproj
 D "Pods/Target Support Files/AFNetworking/AFNetworking-Private.xcconfig"
 M "Pods/Target Support Files/AFNetworking/AFNetworking.xcconfig"
 D "Pods/Target Support Files/FLAnimatedImage/FLAnimatedImage-Private.xcconfig"
 M "Pods/Target Support Files/FLAnimatedImage/FLAnimatedImage.xcconfig"
 D "Pods/Target Support Files/KILabel/KILabel-Private.xcconfig"
 M "Pods/Target Support Files/KILabel/KILabel.xcconfig"
 D "Pods/Target Support Files/MBProgressHUD/MBProgressHUD-Private.xcconfig"
 M "Pods/Target Support Files/MBProgressHUD/MBProgressHUD.xcconfig"
 D "Pods/Target Support Files/Mixpanel/Mixpanel-Private.xcconfig"
 M "Pods/Target Support Files/Mixpanel/Mixpanel.xcconfig"
 D "Pods/Target Support Files/PINCache/PINCache-Private.xcconfig"
 M "Pods/Target Support Files/PINCache/PINCache.xcconfig"
 D "Pods/Target Support Files/PINRemoteImage/PINRemoteImage-Private.xcconfig"
 M "Pods/Target Support Files/PINRemoteImage/PINRemoteImage.xcconfig"
 M "Pods/Target Support Files/Pods-CocoapodsTest/Pods-CocoapodsTest-resources.sh"
 M "Pods/Target Support Files/Pods-CocoapodsTest/Pods-CocoapodsTest.debug.xcconfig"
 M "Pods/Target Support Files/Pods-CocoapodsTest/Pods-CocoapodsTest.release.xcconfig"
 D "Pods/Target Support Files/SVProgressHUD/SVProgressHUD-Private.xcconfig"
 M "Pods/Target Support Files/SVProgressHUD/SVProgressHUD.xcconfig"
 D "Pods/Target Support Files/Stripe/Stripe-Private.xcconfig"
 M "Pods/Target Support Files/Stripe/Stripe.xcconfig"
?? Pods/Target Support Files/Pods-CocoapodsTest/Pods-CocoapodsTest-frameworks.sh
$ git add -A
$ git commit -am "Pod install"
[master 58e70e0] Pod install
 17 files changed, 1919 insertions(+), 1806 deletions(-)
 rename Pods/Target Support Files/AFNetworking/{AFNetworking-Private.xcconfig => AFNetworking.xcconfig} (81%)
 rename Pods/Target Support Files/FLAnimatedImage/{FLAnimatedImage-Private.xcconfig => FLAnimatedImage.xcconfig} (82%)
 rename Pods/Target Support Files/KILabel/{KILabel-Private.xcconfig => KILabel.xcconfig} (88%)
 rename Pods/Target Support Files/MBProgressHUD/{MBProgressHUD-Private.xcconfig => MBProgressHUD.xcconfig} (86%)
 rename Pods/Target Support Files/Mixpanel/{Mixpanel-Private.xcconfig => Mixpanel.xcconfig} (74%)
 rename Pods/Target Support Files/PINCache/{PINCache-Private.xcconfig => PINCache.xcconfig} (87%)
 rename Pods/Target Support Files/PINRemoteImage/{PINRemoteImage-Private.xcconfig => PINRemoteImage.xcconfig} (86%)
 create mode 100755 Pods/Target Support Files/Pods-CocoapodsTest/Pods-CocoapodsTest-frameworks.sh
 rename Pods/Target Support Files/SVProgressHUD/{SVProgressHUD-Private.xcconfig => SVProgressHUD.xcconfig} (86%)
 rename Pods/Target Support Files/Stripe/{Stripe-Private.xcconfig => Stripe.xcconfig} (82%)
$ pod install --no-repo-update
Analyzing dependencies
Downloading dependencies
Using AFNetworking (2.6.0)
Using FLAnimatedImage (1.0.8)
Using Heap (2.1.5)
Using Instabug (4.1.4)
Using KILabel (1.0.1)
Using MBProgressHUD (0.9.1)
Using Mixpanel (2.8.2)
Using PINCache (2.0)
Using PINRemoteImage (1.1.2)
Using SVProgressHUD (1.1.3)
Using Stripe (5.1.2)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 9 dependencies from the Podfile and 11
total pods installed.
$ gs
## master
$ pod install --no-repo-update
Analyzing dependencies
Downloading dependencies
Using AFNetworking (2.6.0)
Using FLAnimatedImage (1.0.8)
Using Heap (2.1.5)
Using Instabug (4.1.4)
Using KILabel (1.0.1)
Using MBProgressHUD (0.9.1)
Using Mixpanel (2.8.2)
Using PINCache (2.0)
Using PINRemoteImage (1.1.2)
Using SVProgressHUD (1.1.3)
Using Stripe (5.1.2)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 9 dependencies from the Podfile and 11
total pods installed.
$ gs
## master
$ pod --version
0.39.0
```

**Comment by segiddins** [comment]
Closing, since this was reported fixed.
